staking/programs/staking/src/state/vesting.rs

```diff
@@ -629,4 +639,147 @@ pub mod tests {
             None
         );
     }
+
+    #[quickcheck]
+    fn test_split_with_args(transferred: u64, total: u64, initial_balance: u64) -> TestResult {
+        if transferred > total || total == 0 {
+            return TestResult::discard();
+        }
+
+        let schedule = VestingSchedule::FullyVested;
+        let (remaining_schedule, transferred_schedule) = schedule
+            .split_vesting_schedule(total - transferred, transferred, total)
+            .unwrap();
+
+        assert_eq!(remaining_schedule, VestingSchedule::FullyVested);
+        assert_eq!(transferred_schedule, VestingSchedule::FullyVested);
+
+        let schedule = PeriodicVesting {
+            initial_balance,
+            // all of these fields should be preserved in the result
+            start_date: 203,
+            period_duration: 100,
+            num_periods: 12,
```

Comment:
Perhaps use the parameters we're going to use in the real vesting schedules for this test

Reply:
I didn't do this one because it makes the invariant test below much more expensive. I don't think the length of the start_date or duration should matter for the splitting logic though, so i think it's better to have the invariant test and not have this.